### PR TITLE
feat(api): handle expiration on MFA session

### PR DESCRIPTION
**Pull request by fsamin**
Signed-off-by: francois  samin <francois.samin@corp.ovh.com>

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#15393.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#15394.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Review by sguiheux — APPROVED** [review]

**Review by yesnault — APPROVED** [review]

File: engine/api/authentication/session.go [hunk 1]
```diff
@@ -12,19 +12,28 @@ import (
...
+		mfaDuration = MFADuration
+	}
+
 	if err := InsertSession(ctx, db, &s); err != nil {
-		return nil, err
+		return nil, mfaDuration, err
 	}
 
-	return &s, nil
+	return &s, mfaDuration, nil
```
Comment (richardlt) [inline engine/api/authentication/session.go:36]: Could be nice to return the computed MFAExpireAt to be able to compute both ExpireAt and MFAExpireAt from the same time.Now() value.

**Review by richardlt — CHANGES_REQUESTED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#15400.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

File: engine/api/authentication/session.go (hunk 1 above)
Comment (fsamin) [reply engine/api/authentication/session.go]: done

**Review by richardlt — APPROVED** [review]

**Pull request merged**